src/utils/config.jl

```diff
@@ -0,0 +1,129 @@
+# TODO (later): move this definition to external files
+const DEFAULT_CONFIG = Dict{String,Any}(
+    "performance" => Dict{String,Any}(
+        "full_analysis" => Dict{String,Any}(
+            "debounce" => 1.0,
+            "throttle" => 5.0
+        )
+    ),
+    "testrunner" => Dict{String,Any}(
+        "executable" => "testrunner",
+    ),
+)
+
+const CONFIG_RELOAD_REQUIRED = Dict{String,Any}(
+    "performance" => Dict{String,Any}(
+        "full_analysis" => Dict{String,Any}(
+            "debounce" => true,
+            "throttle" => true
+        )
+    ),
+    "testrunner" => Dict{String,Any}(
+        "executable" => false,
+    ),
+)
```

Comment:
A quick investigation suggests that registering our configuration schema on platforms like https://www.schemastore.org/ could enable features such as autocompletion and validation for users.
(There also appear to be various easy editor-specific mechanisms for supporting. Of course, this is independent of the language server itself.)

However, this investigation is still incomplete. For now, the default configuration values are hardcoded. As the number of configuration options grows, we will likely move both the values and the schema to an external configuration file, after deciding on a more detailed approach.